### PR TITLE
[master forward-port] Fixed RPMs using artifacts compiled against a too-new version of glibc

### PR DESCRIPTION
**Pull request by fheinecke**
This is a forward port of https://github.com/gravitational/teleport/pull/11008 into master. There are still a couple of issues (namely, this only affects amd64 RPMs and not all architectures), but they should be fixed in a future PR after v9 is released.

### CODE REVIEW

**Review by wadells — APPROVED** [review]

File: build.assets/build-package.sh [hunk 1]
```diff
@@ -192,14 +192,19 @@ elif [[ "${ARCH}" == "arm64" ]]; then
     TEXT_ARCH="ARMv8/ARM64"
 fi
 
+# amd64 RPMs should use CentOS 7 compatible artifacts
+if [[ "${PACKAGE_TYPE}" == "rpm" && "${ARCH}" == "x86_64" ]]; then
+    OPTINAL_TARBALL_SECTION+="-centos7"
```
Comment (webvictim) [inline build.assets/build-package.sh:197]:
```suggestion
    OPTIONAL_TARBALL_SECTION+="-centos7"
```

File: build.assets/build-package.sh [hunk 2]
```diff
@@ -210,7 +215,7 @@ if [[ "${TELEPORT_TYPE}" == "ent" ]]; then
         TYPE_DESCRIPTION="[${TEXT_ARCH} Enterprise edition]"
     fi
 else
-    TARBALL_FILENAME="teleport-v${TELEPORT_VERSION}-${PLATFORM}-${FILENAME_ARCH}${OPTIONAL_RUNTIME_SECTION}-bin.tar.gz"
+    TARBALL_FILENAME="teleport-v${TELEPORT_VERSION}-${PLATFORM}-${FILENAME_ARCH}${OPTINAL_TARBALL_SECTION}${OPTIONAL_RUNTIME_SECTION}-bin.tar.gz"
```
Comment (webvictim) [inline build.assets/build-package.sh:218]:
```suggestion
    TARBALL_FILENAME="teleport-v${TELEPORT_VERSION}-${PLATFORM}-${FILENAME_ARCH}${OPTIONAL_TARBALL_SECTION}${OPTIONAL_RUNTIME_SECTION}-bin.tar.gz"
```

**Review by r0mant — APPROVED** [review]
@fheinecke Can you address comments here and merge?

**Review by espadolini — APPROVED** [review]
LGTM other than the typo that @webvictim commented about

**Review by russjones — APPROVED** [review]

**Comment by r0mant** [comment]
@fheinecke Can you merge this?

File: build.assets/build-package.sh (hunk 2 above)
Comment (fheinecke) [reply build.assets/build-package.sh]: Thanks - missed these

**Pull request merged**